src/v0/destinations/braze/util.js

```diff
@@ -284,12 +284,17 @@ const BrazeDedupUtility = {
         return true;
       });
 
-    if (keys.length === 0) {
-      return null;
+    if (keys.length > 0) {
+      keys.forEach((key) => {
+        if (!_.isEqual(userData[key], storedUserData[key])) {
+          deduplicatedUserData[key] = userData[key];
+        }
+      });
     }
 
-    keys.forEach((key) => {
-      if (!_.isEqual(userData[key], storedUserData[key])) {
+    // add non billable attributes back to the deduplicated user object
+    BRAZE_NON_BILLABLE_ATTRIBUTES.forEach((key) => {
+      if (isDefinedAndNotNull(userData[key])) {
```

Comment:
Should also forward non-billable attributes that are null as braze accepts null attributes ?

Reply:
Addressed